### PR TITLE
clarified authentication note

### LINKED ISSUE #335: Update docs: Batch API always requires to be authenticated
From #334, we've seen that we need to change documentation in order to avoid misunderstandings. For instance, where we say: `In order to use the SQL Batch API, your table must be public, or you must be authenticated using API keys.`. We have to say: `In order to use the SQL Batch API, you must be authenticated using API keys.`. In other words, to use Batch API you'll need always to indicate your `api_key` in the request.

Also we have to change [this section](https://carto.com/docs/carto-engine/sql-api/sql-batch-api/#private-datasets) to reflect these changes.

Thanks!

### PR DESCRIPTION
**Pull request by csobier**
Fixes #335 

@dgaubert , good call.  Please review

### CODE REVIEW

File: doc/sql_batch_api.md [hunk 1]
```diff
@@ -471,7 +471,7 @@ In some scenarios, you may need to fetch the output of a job. If that is the cas
 
 ## Private Datasets
```
Comment (dgaubert) [inline doc/sql_batch_api.md]: It'd be a good idea to change this title and use `Authentication` instead. Also, I'd move this section to the top of this document as first section and remove the note that references to this section.

What do you think?

File: doc/sql_batch_api.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ The SQL Batch API enables you to request queries with long-running CPU processin
 
 _The Batch API is not intended to be used for large query payloads than contain over 4096 characters (4kb). For instance, if you are inserting a large number of rows into your table, you still need to use the [Import API](http://docs.carto.com/carto-engine/import-api/) or [SQL API](http://docs.carto.com/carto-engine/sql-api/) for this type of data management. The Batch API is specific to queries and CPU usage._
 
-**Note:** In order to use the SQL Batch API, your table must be public, or you must be [authenticated](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) using API keys. For details about how to manipulate private datasets with the SQL Batch API, see [Private Datasets](#private-datasets).
+**Note:** In order to use the SQL Batch API, your table must be public, and you **must** be [authenticated](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) using API keys. For details about how to manipulate private datasets with the SQL Batch API, see [Private Datasets](#private-datasets).
```
Comment (dgaubert) [inline doc/sql_batch_api.md]: For Batch API there is no differences between access to public or private data. So I would say: 
```
Note: In order to use the SQL Batch API you must be authenticated using API keys. For details about how to work with the SQL Batch API, see Authentication.
```

File: doc/sql_batch_api.md [hunk 3]
```diff
@@ -471,7 +471,7 @@ In some scenarios, you may need to fetch the output of a job. If that is the cas
 
 ## Private Datasets
 
-For access to all private tables, and for write access to public tables, an API Key is required to [authenticate](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) your queries with the Batch API. The following error message appears if you are using private tables and are not authenticated:
+For access to all private tables, and for write access to public tables, an **API Key is required** to [authenticate](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) your queries with the Batch API. The following error message appears if you are using private tables and are not authenticated:
```
Comment (dgaubert) [inline doc/sql_batch_api.md]: Same here, I would say:
`An API Key is required to manage your jobs. The following error message appears if you are not authenticated:`

File: doc/making_calls.md [hunk 4]
```diff
@@ -37,7 +37,7 @@ https://{username}.carto.com/api/v2/sql?q=SELECT count(*) FROM {table_name}
 }
 ```
 
-Finally, remember that in order to use the SQL API, either your table must be public, or you must be [authenticated](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) using API Keys.
+Finally, remember that in order to use the SQL API, your table must be public, and you **must** be [authenticated](http://docs.carto.com/carto-engine/sql-api/authentication/#authentication) using API Keys.
```
Comment (dgaubert) [inline doc/making_calls.md]: There is no need to change anything in this document, just for the Batch API. 

Batch API is an extension of SQL API, but they aren't the same thing.

Comment (csobier) [inline doc/making_calls.md]: Ah, okay. Didn't realize it was Batch specific, reverting.

**Comment by csobier** [comment]
I applied @dgaubert edits, looks good I think?

**Comment by dgaubert** [comment]
👍 

Let's merge & ship!

**Pull request merged**